### ISSUE TITLE
reraise_exceptions

### ISSUE THREAD

**Issue by kitchoi** [enthought/traitsui#1232]

**Comment by kitchoi** [comment]
After #1231 is merged and after moving `push_exception_handler`/`pop_exception_handler` to test case setup/teardown, tests that use both `reraise_exceptions()` and `create_ui` like this:
https://github.com/enthought/traitsui/blob/80a31705d974edd9624ceb34cd3e83a6f643960a/traitsui/tests/editors/test_drop_editor.py#L47-L48

can just be `with create_ui(...): ...`

**Comment by aaronayres35** [comment]
> After #1231 is merged and after moving `push_exception_handler`/`pop_exception_handler` to test case setup/teardown, tests that use both `reraise_exceptions()` and `create_ui` like this:
> https://github.com/enthought/traitsui/blob/80a31705d974edd9624ceb34cd3e83a6f643960a/traitsui/tests/editors/test_drop_editor.py#L47-L48
> 
> can just be `with create_ui(...): ...`

Once #1270 is merged should we open a separate issue to make all these replacements?

**Comment by kitchoi** [comment]
I think what I said there was too simplistic. My apologies. I had a look at the occurrences of `reraise_exceptions` we have.
Some of these `reraise_exceptions` are now obviously redundant. But some do need to stay because not all test logics go via UITester (strictly speak, UIWrapper) [where we are wrapping GUI event processing with the same context manager](https://github.com/enthought/traitsui/blob/aa16c9cabc6d4d4904f30103f09b32227a8199fc/traitsui/testing/tester/ui_wrapper.py#L322-L332).

It costs little extra to keep the redundant ones, apart from the fact that they are now noise to developers reading the test code. To me it isn't important enough to say there is an issue about keeping them, but that could be just me.
